config/config-feature-flags.yaml

```diff
@@ -40,14 +40,23 @@ data:
   # See https://github.com/tektoncd/pipeline/issues/2013 for more
   # info.
   disable-home-env-overwrite: "false"
-  # Setting this flag to "true" will prevent Tekton overriding your
-  # Task container's working directory.
+  # Setting this flag to "false" will enforce TaskRun's timeout
+  # check to Step level enforcement.
   #
-  # The default behaviour currently is for Tekton to override the
-  # working directory if not set by the user but this will change
-  # in an upcoming release.
+  # The default behaviour currently is for the TaskRun's timeout
+  # check is done at TaskRun's reconciler level insteaf of at
+  # Step level.
+  #
+  # See https://github.com/tektoncd/pipeline/issues/2559 for more
+  # info.
+  disable-taskrun-step-timeout-check: "true"
```

Comment:
Suggest making this `enforce-taskrun-timeouts-in-step-entrypoint: "false"` to avoid the double-negative that occurs when `disable` and `false` together.

We've used `disable-` for many of the other feature flags mostly because they turn off features which existed before the feature flags were introduced.  Here though I think we're adding a new implementation and offering users the opportunity to test it out before we switch over to it permanently.